pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageImpl.java

```diff
@@ -319,10 +321,17 @@ public boolean publishedEarlierThan(long timestamp) {
         }
     }
 
-    public Schema<T> getSchema() {
+    public Schema<T> getCurrentSchema() {
         return this.schema;
     }
 
+    public CompletableFuture<Schema<?>> getOriginalSchema() {
```

Comment:
can you please add a javadoc that explains why method returns a CompletableFuture ?
for instance we could need to download the schema from the network.

can we name this method "fetchSchema" ?
This way the name itself reads that this is not a simple "getter" but that the operation may perform additional operations.